core/src/main/java/hudson/model/ChoiceParameterDefinition.java

```diff
@@ -51,6 +55,60 @@ private ChoiceParameterDefinition(String name, List<String> choices, String defa
         this.defaultValue = defaultValue;
     }
 
+    /**
+     * Databound constructor for reflective instantiation.
+     *
+     * @param name parameter name
+     * @param description parameter description
+     *
+     * @since TODO
+     */
+    @DataBoundConstructor
+    @Restricted(NoExternalUse.class) // there are specific constructors with String and List arguments for 'choices'
+    public ChoiceParameterDefinition(String name, String description) {
+        super(name, description);
+        this.choices = new ArrayList<>();
+        this.defaultValue = null;
+    }
+
+    /**
+     * Set the list of choices. Legal arguments are String (in which case the arguments gets split into lines) and Collection
+     * which sets the list of legal parameters to the String representations of the argument's non-null entries.
+     *
+     * See JENKINS-26143 for background.
+     *
+     * This retains the compatibility with the legacy String 'choices' parameter, while supporting the list type as generated
+     * by the snippet generator.
+     *
+     * @param choices String or Collection representing this parameter definition's possible values.
+     *
+     * @since TODO
+     *
+     */
+    @DataBoundSetter
+    @Restricted(NoExternalUse.class) // this is terrible enough without being used anywhere
+    public void setChoices(Object choices) {
```

Comment:
This seems to break [this test](https://github.com/jenkinsci/workflow-cps-plugin/blob/master/src/test/java/org/jenkinsci/plugins/workflow/cps/SnippetizerTest.java#L335).